lib/matplotlib/dates.py

```diff
@@ -1161,10 +1161,16 @@ def nonsingular(self, vmin, vmax):
 class RRuleLocator(DateLocator):
```

Comment:
Sorry, I should have noticed this the first time.  You have also changed the signature of RRuleLocator.  I think you need to write a new docstring for it, since it cannot simply inherit from `DateLocator` any longer.

Reply:
I'm actually not completely happy with that signature change yet. Calling the argument 'interval_multiples' is a bit confusing if it really only applies to yearly ticks. 

Taking care of the special case directly in the RRuleLocator is best for overall clarity, imo. That's why I think this change is an improvement. 

But I also want to achieve easy usage of the RRuleLocator for timedeltas at the same time. It already works for all frequencies except for daily. Daily with timedeltas is a very similar special case to yearly with dates. I'd like to subclass the RRuleLocator and only override init and the method that determines start, stop as multiples of the interval.

It might be better to have the signature be different between RRuleLocator and a subclass to make everything more clear?
For example, call the argument 'yearly_interval_multiples' maybe?

It would be best to have only one RRuleLocator class for dates and timedeltas. Also because of the locator classes for specific frequencies. As it looks, HourLocator can be used for dates and timedeltas. But DayLocator cannot be used for both because that is a special case with timedeltas. Ideally, all locators can be used for both. But I see no way currently to make that happen.

This is, of course, out of scope for this PR. But I do want to consider it already to facilitate the other PR.

Reply:
If it really can't be done generally, then maybe a subClass make sense.  In that case you could make it the same as the previous YearlyLocator, but hopefully with less repeated code because you now subclass RRuleLocator.  That seems the most straightforward, and it stops RRuleLocator from having a bunch of extra logic?

Reply:
That might actually be the best option. It still requires overwriting the `.tick_values` method in the yearly locator. There will be some very similar but slightly different code then. But less complex than the previous YearLocator. So an improvement after all. 
And it keeps the extra logic out of the RRuleLocator, that's true. And it's not changing the signature of RRuleLocator.

But it also means that this has to stay in `AutoDateLocator.get_locator()`:
```
  if (freq == YEARLY) and self.interval_multiples:
      locator = YearLocator(interval, tz=self.tz)
  elif use_rrule_locator[i]:
      ...
      locator = RRuleLocator(rrule, self.tz)
```
I don't particularly like this because it is not very flexible, considering an AutoTimedeltaLocator requires a different special case here. But maybe that's not a problem to solve here.